### PR TITLE
Readme update aql examples

### PR DESCRIPTION
**Pull request by rajivsam**
1. Updated README with installation pre-requisites
2. Added AQL examples. Screen shots and queries were added. 
3. The testcases in arangopipe_testcases now take a config object. To run the tests, edit the run_tests() method in arangopipe_testcases.py. Specify the connection parameters pertainining to your ArangoDB installation in the set_dbconnection() method.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,3 +51,31 @@ Arangopipe represents metadata as a graph. The nodes of the graph above are the
...
+
+## Installing Arangopipe
+
+This repository contains **Arangopipe** and examples to illustrate how it can be used with machine learning tools like **hyperopt** and **MLFlow**. To install **Arangopipe**, do the following:
+
+1.  Install ArangoDB
+
+    `docker run -p 8529:8529 -e ARANGO_ROOT_PASSWORD=openSesame arangodb/arangodb:3.4.6`
+    
+2.  Install python-arango
```
Comment (joerg84) [inline README.md]: 2. Install prerequisites

`pip install python-arango PyYaml`

and then we can drop 3.

File: arangopipe/tests/AQL/query_1.json [hunk 2]
```diff
@@ -0,0 +1 @@
...rmations'\n        REMOVE f IN featuresets","parameter":{},"name":"query_2"},{"value":"INSERT {\"name\" : \"wine dataset\",\n                   \"description\": \"Wine quality ratings\",\"source\": \"UCI ML Repository\" } INTO datasets","parameter":{},"name":"query_3"},{"value":"INSERT {\"fixed acidity\": \"float64\",\n               \"volatile acidity\": \"float64\",\n               \"citric acid\": \"float64\",\n               \"residual sugar\": \"float64\",\n               \"chlorides\": \"float64\",\n               \"free sulfur dioxide\": \"float64\",\n               \"total sulfur dioxide\": \"float64\",\n               \"density\": \"float64\",\n               \"pH\": \"float64\",\n               \"sulphates\": \"float64\",\n               \"alcohol\": \"float64\",\n               \"quality\": \"int64\",\n               \"name\": \"wine_no_transformations\"} INTO featuresets","parameter":{},"name":"query_4"},{"value":"FOR ds IN datasets\n    FILTER ds.name == 'wine dataset'\n        FOR fs in featuresets\n            FILTER fs.name == 'wine_no_transformations'\n            INSERT {\"_from\": fs._id, \"_to\": ds._id} INTO featureset_dataset","parameter":{},"name":"query_5"}]
```
Comment (joerg84) [inline arangopipe/tests/AQL/query_1.json]: Does it make sense to have a markdown file describing these queries? Or am I missing this file (as you have the graphics...)

File: README.md (hunk 1 above)
Comment (joerg84) [reply README.md]: also we should probably fix the versions here...
Maybe use a `pip install -r requirements.txt` ?

Comment (rajivsam) [reply README.md]: Agree.

File: arangopipe/tests/AQL/query_1.json (hunk 2 above)
Comment (rajivsam) [reply arangopipe/tests/AQL/query_1.json]: Yes, good idea.

**Comment by rajivsam** [comment]
Will make the suggested changes and revert

**Comment by joerg84** [comment]
Why do you close the PR?

**Comment by rajivsam** [comment]
> Why do you close the PR?

My mistake. I should simply have pushed the new changes here. Sorry about that.

File: arangopipe/tests/AQL/Running_AQL.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Running AQL queries  on Arangopipe
```
Comment (joerg84) [inline arangopipe/tests/AQL/Running_AQL.md]: 2 spaces between `queries` and `on`

File: arangopipe/tests/AQL/Running_AQL.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+1. Deleting a dataset from the database
+2. Deleting a featureset from the database
+3. Inserting a dataset into the database
+4. Inserting a featureset into the database
+5. Linking the featureset and the dataset
+
+If you would like to learn more about AQL, please view the [AQL documentation](https://www.arangodb.com/docs/stable/aql/). The [tutorials](https://www.arangodb.com/docs/stable/tutorials.html) section of the documentation may also be of interest.
+
+## Deleting a dataset from the database
```
Comment (joerg84) [inline arangopipe/tests/AQL/Running_AQL.md]: Just curious: why do we show a delete first, IMO that should only be rare occasions as you destroy an audit trail.

File: arangopipe/tests/AQL/Running_AQL.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    REMOVE d IN datasets
+  
+```
+This query can be executed by entering the query in the query editor and execute the query by clicking the **_Execute_** button (see Figure 1).
+
+## Deleting a featureset from the database
+
+A featureset can be deleted from the database in a similar manner. The AQL query to delete a featureset vertex with a particular name is shown below.
+
+```
```
Comment (joerg84) [inline arangopipe/tests/AQL/Running_AQL.md]: Dito

File: arangopipe/tests/AQL/Running_AQL.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
+# Running AQL queries  on Arangopipe
+
+Apart, from using the **Arangopipe** search feature, it is also possible to query machine learning meta-data using the query feature in **ArangoDB**. An excerpt of the feature is shown in the figure below.
+<figure>
+  <img src="AQL_feature.png" alt="my alt text" width=90%/>
+  <figcaption style ="text-align:center">Figure 1.</figcaption>
+</figure>
+The left pane shows an editor area where we can enter queries. We can the execute the query by clicking on the *_Execute_* button in the lower right hand corner. Let us illustrate this with an example. The dataset used for illustration is the wine dataset used to illustrate how **Arangopipe** can be used with *_mlflow_* (see the *_mlflow_* folder). We will illustrate the following operations:
+
+1. Deleting a dataset from the database
```
Comment (joerg84) [inline arangopipe/tests/AQL/Running_AQL.md]: Should we maybe add a graph traversal query and not just DDL?

File: requirements.txt [hunk 7]
```diff
@@ -0,0 +1,2 @@
+python-arango
```
Comment (joerg84) [inline requirements.txt]: Do you want to ping those to specific versions? E.g., by `pip freeze -r requirements.txt > stable-req.txt`

File: arangopipe/tests/AQL/Running_AQL.md (hunk 3 above)
Comment (rajivsam) [reply arangopipe/tests/AQL/Running_AQL.md]: Sure, will take care of this.

File: arangopipe/tests/AQL/Running_AQL.md (hunk 4 above)
Comment (rajivsam) [reply arangopipe/tests/AQL/Running_AQL.md]: Doing the delete first makes no assumptions about the state of the database while executing the actions in the sequence listed in the markdown file. If the user had executed this exercise a few times, the inserted results will be in the database. This sequence gaurantees the same results each time and serves to illustrate basic CRUD operations.

File: arangopipe/tests/AQL/Running_AQL.md (hunk 5 above)
Comment (rajivsam) [reply arangopipe/tests/AQL/Running_AQL.md]: No, problem. I will move this also to the end.

File: arangopipe/tests/AQL/Running_AQL.md (hunk 6 above)
Comment (rajivsam) [reply arangopipe/tests/AQL/Running_AQL.md]: Sure, I just need to add some cases to insert some run information into the database before querying it. Yeah, that makes that complete.

File: requirements.txt (hunk 7 above)
Comment (rajivsam) [reply requirements.txt]: yes, that makes sense.

File: arangopipe/tests/AQL/Running_AQL.md (hunk 4 above)
Comment (joerg84) [reply arangopipe/tests/AQL/Running_AQL.md]: IMO delete destroys a potential audit trail, so maybe we should even consider whether that needs a special role/permissions from the arangoml UI (just documenting a thought here ;-) )

**Pull request merged**